src/html5/parser.rs

```diff
@@ -1,5 +1,6 @@
 mod attr_replacements;
 pub mod document;
+pub mod query;
```

Comment:
It vaguely looks like the `cargo bench` results are different between this branch and `main`.  I don't have a theory as to what's going on, since this branch seems purely additive.  Perhaps the benchmarks are flawed?

```sh
$ git checkout main
$ cargo bench # three times
$ git checkout feat/tree-traversal
$ cargo bench # three times
# See file:///home/walker/code/emwalker/gosub-engine/target/criterion/report/index.html (in my case)
```

![2023-11-09_16-49](https://github.com/gosub-browser/gosub-engine/assets/760949/ac94209a-a388-4bc9-b160-505ae54f8d2b)

![2023-11-09_16-46](https://github.com/gosub-browser/gosub-engine/assets/760949/8141c179-d3e5-4417-9d6e-892f6ed90da4)

Reply:
yeah that’s odd, I’m not sure what that would be caused by since I didn’t modify any of the existing implementations, only added a couple methods on the doc itself along with the new structs

Reply:
Can you double-check and see if you see the same thing?

Reply:
Ok, so something weird must be happening on my system.  Thanks for checking.